hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/ArtifactManagementTest.java

```diff
@@ -173,14 +173,70 @@ public void createArtifactsUntilQuotaIsExceeded() throws NoSuchAlgorithmExceptio
                 .isEqualTo(maxArtifacts - 1);
 
         // now we should be able to create an artifact again
-        final byte random[] = RandomStringUtils.random(artifactSize).getBytes();
-        try (final InputStream inputStream = new ByteArrayInputStream(random)) {
+        final byte[] randomBytes = randomBytes(artifactSize);
+        try (final InputStream inputStream = new ByteArrayInputStream(randomBytes)) {
             artifactManagement.create(inputStream, sm1.getId(), "fileXYZ", false, artifactSize);
             assertThat(artifactRepository.findBySoftwareModuleId(PAGE, sm1.getId()).getTotalElements())
                     .isEqualTo(maxArtifacts);
         }
     }
 
+    @Test
+    @Description("Verifies that the quota specifying the maximum artifact storage is enforced (across software modules).")
+    public void createArtifactsUntilStorageQuotaIsExceeded() throws NoSuchAlgorithmException, IOException {
+
+        // create as many artifacts as possible w/o violating the storage quota
+        final long maxBytes = quotaManagement.getMaxArtifactStorage();
+        final List<Long> artifactIds = Lists.newArrayList();
+        final int artifactSize = 320 * 1024;
+        final int numArtifacts = Math.toIntExact(maxBytes / artifactSize);
+        for (int i = 0; i < numArtifacts; ++i) {
+            final JpaSoftwareModule sm = softwareModuleRepository
+                    .save(new JpaSoftwareModule(osType, "smd" + i, "1.0", null, null));
+            final byte[] randomBytes = randomBytes(artifactSize);
+            try (final InputStream inputStream = new ByteArrayInputStream(randomBytes)) {
+                artifactIds.add(
+                        artifactManagement.create(inputStream, sm.getId(), "file" + i, false, artifactSize).getId());
+            }
+        }
+
+        // upload one more artifact to trigger the quota exceeded error
+        final JpaSoftwareModule sm = softwareModuleRepository
+                .save(new JpaSoftwareModule(osType, "smd" + numArtifacts, "1.0", null, null));
+        assertThatExceptionOfType(QuotaExceededException.class).isThrownBy(() -> {
+            final byte[] randomBytes = randomBytes(artifactSize);
+            try (final InputStream inputStream = new ByteArrayInputStream(randomBytes)) {
+                artifactManagement.create(inputStream, sm.getId(), "file" + numArtifacts, false, artifactSize);
+            }
+        });
+
+        // delete one of the artifacts
+        artifactManagement.delete(artifactIds.get(0));
+
+        // now we should be able to create an artifact again
+        final byte[] randomBytes = RandomStringUtils.randomAlphanumeric(artifactSize).getBytes();
```

Comment:
use the randomBytes() method that you defined

Reply:
done